src/main/java/com/atlassian/bitbucket/jenkins/internal/http/HttpRequestExecutorImpl.java

```diff
@@ -132,6 +111,54 @@ private static void handleError(int responseCode, @Nullable String body)
         throw new UnhandledErrorException("Unhandled error", responseCode, body);
     }
 
+    private <T> T performRequest(Request request, ResponseConsumer<T> consumer) {
+        try {
+            Response response = httpCallFactory.newCall(request).execute();
+            int responseCode = response.code();
+
+            try (ResponseBody body = response.body()) {
+                if (response.isSuccessful()) {
+                    log.fine("Bitbucket - call successful");
+                    return consumer.consume(response);
+                }
+                handleError(responseCode, body == null ? null : body.string(), response.headers());
+            }
+        } catch (ConnectException | SocketTimeoutException e) {
+            log.log(Level.FINE, "Bitbucket - Connection failed", e);
+            throw new ConnectionFailureException(e);
+        } catch (IOException e) {
+            log.log(Level.FINE, "Bitbucket - io exception", e);
+            throw new BitbucketClientException(e);
+        } catch (RateLimitedException e) {
+            RetryOnRateLimitConfig rateLimitConfig = request.tag(RetryOnRateLimitConfig.class);
+            if (rateLimitConfig != null) {
+
+                if (rateLimitConfig.incrementAndGetAttempts() <= rateLimitConfig.getMaxAttempts()) {
+                    try {
+                        Thread.sleep(e.getRetryIn());
+                    } catch (InterruptedException ex) {
+                        throw new UnhandledErrorException("Interrupted during wait to retry", -2, null);
```

Comment:
Are these response codes translated anywhere? Why `-2`? (and why `-1` below?)

Reply:
They are unique, they are logged elsewhere (typically), a negative one indicates that it was an error that is not related to the remote server, the number is just a unique counter to help locate it.